app/domain/authentication/authn_jwt/fetch_mandatory_claims_to_validate.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnJwt
+    OPTIONAL_CLAIMS = [ISS_CLAIM_NAME, EXP_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME].freeze
+
+    # FetchMandatoryClaimsToValidate command class is responsible to return a list of mandatory JWT standard claims to
+    # validate.
+    FetchMandatoryClaimsToValidate ||= CommandClass.new(
+      dependencies: {
+        fetch_issuer_value: ::Authentication::AuthnJwt::FetchIssuerValue.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authentication_parameters]
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwtMandatoryClaimsToValidate.new)
+        fetch_mandatory_claims_to_validate
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchedJwtMandatoryClaimsToValidate.new)
+
+        @mandatory_claims_to_validate_list
+      end
+
+      private
+
+      # fetch_mandatory_claims_to_validate function is responsible to return a list of mandatory JWT standard claims to
+      # validate, according to the following logic:
+      # For each optional claims (iss , exp,nbf, iat) which exists in the token, add to mandatory list
+      # Note: mandatory list also contains the value to validate if necessary (for example iss: cyberark.com)
+      def fetch_mandatory_claims_to_validate
```

Comment:
i don't quite understand the logic here. the function name is `fetch_mandatory_claims_to_validate` and then you iterate over the `OPTIONAL_CLAIMS` list and fetch them. so are you fetching mandatory or optional claims?

Reply:
the following claims: (iss , exp, nbf, iat) are optional in JWT token according to the RFC
The logic is :
1. If they exist in the Token validate them -> mandatory to validate 
2. If they no exist in the token ignore them -> will not appear in the mandatory list 


Note: the `exp` claim may be changed to be mandatory and not optional, so if it not exists in the token we will failed the request